orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -551,9 +553,18 @@ Status OrtLoadInternal(const PathString& checkpoint_path,
     LoadTensorProtoFromFile(tensor_file_full_path, tensor_protos, "[params]");
 
     for (const auto& tensor_proto : tensor_protos) {
-      param_tensor_protos.push_back(tensor_proto);
+      param_tensor_protos.insert(std::make_pair(tensor_proto.name(), std::move(tensor_proto)));
```

Comment:
```suggestion
      param_tensor_protos.emplace(tensor_proto.name(), std::move(tensor_proto));
```

`insert` and `make_pair` will make a copy. Please use emplace instead.